db-service/test/cqn4sql/UPDATE.test.js

```diff
@@ -70,17 +70,52 @@ describe('UPDATE', () => {
     })
   })
 
-  // we do not really understand a token stream such as a where clause,
-  // hence we cannot easily rewrite a path expression into a `where exists` subquery
-  // for the moment, we should issue a proper error instead of dumping.
-  it('Update with join clause is rejected', () => {
+  it('Update with path expressions in where is handled', () => {
     const { UPDATE } = cds.ql
-    let u = UPDATE.entity('bookshop.Books').where(
+    let u = UPDATE.entity({ ref: ['bookshop.Books'] }).where(
       `author.name LIKE '%Bron%' or ( author.name LIKE '%King' and title = 'The Dark Tower') and stock >= 15`,
     )
-    expect(() => cqn4sql(u)).to.throw(
-      'Path expressions for UPDATE statements are not supported. Use “where exists” with infix filters instead.',
+    // this is how it looks internally in cqn4sql before the subquery is transformed
+    let intermediate = UPDATE.entity('bookshop.Books').where(`
+      exists (SELECT 1 from bookshop.Books as Books
+              where author.name LIKE '%Bron%' or ( author.name LIKE '%King' and title = 'The Dark Tower') and stock >= 15
+            )
+    `)
+    let expected = UPDATE.entity({ ref: ['bookshop.Books'] }).where(`
+    exists (SELECT 1 from bookshop.Books as Books
+              left join bookshop.Authors as author on author.ID = Books.author_ID
+              where author.name LIKE '%Bron%' or ( author.name LIKE '%King' and Books.title = 'The Dark Tower') and Books.stock >= 15
+            )
+  `)
```

Comment:
I did some manual, end-to-end testing with delete statements. It seems that an important peace is missing. Without the correlation to the outer query (`UPDATE` in this case), all entries will be matched and hence updated :(

Reply:
My expectation would be something like this
```suggestion
    let expected = UPDATE.entity({ ref: ['bookshop.Books'] }).where(`
    ID IN (SELECT ID from bookshop.Books as Books
              left join bookshop.Authors as author on author.ID = Books.author_ID
              where author.name LIKE '%Bron%' or ( author.name LIKE '%King' and Books.title = 'The Dark Tower') and Books.stock >= 15
            )
  `)
```

Reply:
cool idea, locally, I solved it by injecting the correlation into the `where` of the subquery

Reply:
> Please also add a test for updating a projection. It's important that the SELECT is on the projection while the UPDATE's projection must be resolved to the table below.

what exactly does it mean? Does `cqn4sql` then learn what `resolveView` currently does, also w.r.t. renamings?

Reply:
No, that would simply cover the end-to-end scenario to work. Should be a test outside cqn4sql.